.github/workflows/test-workflow.yml

```diff
@@ -40,6 +40,25 @@ jobs:
       - name: Install dependencies
         run: npm ci
 
+      - name: Install Foundry
```

Comment:
I'd prefer this broken out as if there's ever a failure in Hardhat configs or local node the Foundry part wouldn't get run and if Foundry failed it would make it look like previous tests failed.
Let's just use the new workflow you added and think of how to combine them later.

Reply:
Ok, it's currently setup such that `forge compile` will attempt to compile `./contracts` which depend on dependencies installed in the `node_modules`, so I'll move this to execute once after a given job.

Reply:
I removed this, as `foundry_test.yml` suffices